File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/uncomment/direct/Java/C++/s221078931_commented.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class IntegralRect {
public:
    int height;
    int width;

    IntegralRect(int h, int w) : height(h), width(w) {}

    int diagonal() const {
        if (height >= width) {
            return INT_MAX;
        }
        return height * height + width * width;
    }

    bool operator<(const IntegralRect& other) const {
        int d1 = this->diagonal();
        int d2 = other.diagonal();
        if (d1 == d2) {
            return this->height < other.height;
        }
        return d1 < d2;
    }
};

int main() {
    vector<IntegralRect> rects(40000);
    for (int i = 0; i < 200; ++i) {
        for (int j = 0; j < 200; ++j) {
            rects[i * 200 + j] = IntegralRect(i + 1, j + 1);
        }
    }
    sort(rects.begin(), rects.end());

    // debug
    /*
    for (int i = 0; i < 100; ++i) {
        cout << rects[i].height << " " << rects[i].width << " " << rects[i].diagonal() << endl;
    }
    */

    while (true) {
        int h, w;
        cin >> h >> w;
        if (h == 0 && w == 0) {
            break;
        }

        IntegralRect tmpRect(h, w);
        int diag = tmpRect.diagonal();

        for (size_t i = 0; i < rects.size(); ++i) {
            if (rects[i] > tmpRect) {
                cout << rects[i].height << " " << rects[i].width << endl;
                break;
            }
        }
    }
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/Go/s221078931_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type IntegralRect struct {
	height int
	width  int
}

func (r *IntegralRect) diagonal() int {
	if r.height >= r.width {
		return int(^uint(0) >> 1)
	}
	return r.height*r.height + r.width*r.width
}

func (r *IntegralRect) compareTo(o *IntegralRect) int {
	d1 := r.diagonal()
	d2 := o.diagonal()
	if d1 == d2 {
		return r.height - o.height
	}
	return d1 - d2
}

func main() {
	var rects [40000]IntegralRect
	for i := 0; i < 200; i++ {
		for j := 0; j < 200; j++ {
			rects[i*200+j] = IntegralRect{i + 1, j + 1}
		}
	}

	reader := bufio.NewReader(os.Stdin)
	for {
		line, _ := reader.ReadString('\n')
		line = strings.TrimSpace(line)
		if line == "0 0" {
			break
		}
		inputs := strings.Split(line, " ")
		h, _ := strconv.Atoi(inputs[0])
		w, _ := strconv.Atoi(inputs[1])

		tmpRect := IntegralRect{h, w}
		diag := tmpRect.diagonal()

		for _, rect := range rects {
			if rect.compareTo(&tmpRect) > 0 {
				fmt.Printf("%d %d\n", rect.height, rect.width)
				break
			}
		}
	}
}

// End of Code
